src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/RazorLogHubTraceProvider.cs

```diff
@@ -37,7 +37,7 @@ public async Task<TraceSource> InitializeTraceAsync(string logIdentifier, int lo
                 logName: $"{logIdentifier}.{logHubSessionId}",
                 serviceId: new ServiceMoniker($"Razor.{logIdentifier}"));
 
-            using var traceConfig = await TraceConfiguration.CreateTraceConfigurationInstanceAsync(_serviceBroker, cancellationToken).ConfigureAwait(false);
+            using var traceConfig = await LogHub.TraceConfiguration.CreateTraceConfigurationInstanceAsync(_serviceBroker, cancellationToken).ConfigureAwait(false);
             var traceSource = await traceConfig.RegisterLogSourceAsync(_logId, _logOptions, cancellationToken).ConfigureAwait(false);
 
             // Trace source(s) have the debug output in VS as a default listener causing excessive noise.
```

Comment:
There are a few changes to the base library that allow some cleanup here.
- The trace source returned here will have the default listener removed. so there is no need to do this: `traceSource.Listeners.Remove("Default");`
- The SourceLevels setting can be requested in `LoggerOptions`. Setting it in `LoggerOptions` allows it to be overridden elsewhere when requested. The line below, `traceSource.Switch.Level = SourceLevels.ActivityTracing | SourceLevels.Information;`, can be removed and replaced with:
```
private static readonly LoggerOptions _logOptions = new(
            requestedLoggingLevel: new LoggingLevelSettings(SourceLevels.Warning | SourceLevels.ActivityTracing),
            privacySetting: PrivacyFlags.MayContainPersonallyIdentifibleInformation | PrivacyFlags.MayContainPrivateInformation);
```

Reply:
Thanks @kelltrick! Was the change from `SourceLevels.Information` to `SourceLevels.Warning` intentional?

Reply:
Also cc/ @CyrusNajmabadi as Roslyn may need these updates as well 😄